src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray.cs

```diff
@@ -106,14 +106,18 @@ public static ImmutableArray<T> CreateRange<T>(IEnumerable<T> items)
             var immutableArray = items as IImmutableArray;
             if (immutableArray != null)
             {
-                immutableArray.ThrowInvalidOperationIfNotInitialized();
+                Array array = immutableArray.Array;
+                if (array == null)
+                {
+                    throw new InvalidOperationException(SR.InvalidOperationOnDefaultArray);
+                }
```

Comment:
@AArnott Interface calls are relatively expensive, and this one was in a commonly-used codepath and trivial to remove. Plus, it let us delete code. I'll see if I can post some stats on how/if this improves perf.